instrumentation/jmx-metrics/library/src/main/java/io/opentelemetry/instrumentation/jmx/engine/BeanPack.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.jmx.engine;
+
+import javax.management.ObjectName;
+import javax.management.QueryExp;
+
+/**
+ * A class describing a set of MBeans which can be used to collect values for a metric. Objects of
+ * this class are immutable.
+ */
+public class BeanPack {
```

Comment:
i'm not following the association with the word Pack?

Reply:
Yes, I took some liberties with naming ...
The class represents a collection of MBeans, but it is not an arbitrary set. There's an important affinity between the individual items. It is not just that they came from one rule, they are all supposed to share at least one MBean attribute. To stress this affinity, I used _pack_ as in _battery pack_ or _pack of wolves_, as they all work together to generate values for the same metric(s).
An alternative could be `BeanGroup`, if you find it more agreeable.

Reply:
> I used pack as in battery pack or pack of wolves

I like this, but maybe `BeanGroup` is a bit more self-explanatory

Reply:
I keep reading `BeanPack` similar to `MessagePack`, i.e. some "packed" representation of a bean

Reply:
Renamed to BeanGroup